crates/mun_hir/src/ty/tests.rs

```diff
@@ -144,6 +144,21 @@ fn infer_break() {
     )
 }
 
+#[test]
+fn infer_while() {
+    infer_snapshot(
+        r#"
+    fn foo() {
+        let n = 0;
+        while n < 3 { n += 1; };
+        while n < 3 { n += 1; break; };
+        while n < 3 { break 3; };   // Error: break with value in while
```

Comment:
Should this match the actual error?
`break` with value can only appear in a `loop`